### PR TITLE
fix(message): call onClose after removeNotice

### LINKED ISSUE #38524: message.loading 当duration设置为0时，手动关闭无法触发onclose
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/antd-reproduction-template-forked-0o3hz8?file=/index.js)

### Steps to reproduce

点击按钮

### What is expected?

控制台打印need log close msg与message close

### What is actually happening?

只打印了need log close msg

| Environment | Info |
| --- | --- |
| antd | 4.23.1 |
| React | 18.2 |
| System | window 11 |
| Browser | electron |

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [X] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

fix #38524 

### 💡 Background and solution

should call onClose after removeNotice

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | should call onClose after removeNotice |
| 🇨🇳 Chinese | 移除消息后应该调用一下`onClose`方法 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

### CODE REVIEW

File: components/message/demo/loading.md [hunk 1]
```diff
@@ -18,7 +18,9 @@ import { Button, message } from 'antd';
 import React from 'react';
 
 const success = () => {
-  const hide = message.loading('Action in progress..', 0);
+  const hide = message.loading('Action in progress..', 0, () => {
```
Comment (yoyo837) [inline components/message/demo/loading.md:21]: 在补个非0的?

Comment (kiner-tang) [reply components/message/demo/loading.md:21]: 好

Comment (kiner-tang) [reply components/message/demo/loading.md:21]: done

**Comment by kiner-tang** [comment]
看上去不是这一次的pr导致的报错
image

File: components/message/demo/loading.md [hunk 2]
```diff
@@ -14,16 +14,29 @@ title:
...
+  message.loading('Action in progress..', 3, () => {
+    message.success('loading had closed');
+  });
+};
+
+const App: React.FC = () => (
+  <Space>
+    <Button onClick={success}>Display a loading indicator</Button>
+    <Button onClick={success2}>Display a loading and auto hide</Button>
+  </Space>
```
Comment (yoyo837) [inline components/message/demo/loading.md:38]: 非0 的也给一个用例

Comment (kiner-tang) [reply components/message/demo/loading.md:38]: 第2个就是非0的呀，3秒后自动隐藏

Comment (kiner-tang) [reply components/message/demo/loading.md:38]: 哦，明白了，是要加一个测试用例是吧

Comment (kiner-tang) [reply components/message/demo/loading.md:38]: done

**Review by yoyo837 — APPROVED** [review]

**Pull request merged**